Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -510,6 +546,7 @@ WebPageProxy::WebPageProxy(PageClient& pageClient, WebProcessProxy& process, Ref
 #if ENABLE(FULLSCREEN_API)
     , m_fullscreenClient(makeUnique<API::FullscreenClient>())
 #endif
+    , m_processActivityState(makeUniqueRef<ProcessActivityState>(*this))
```

Comment:
Do we really need to use a UniqueRef here? Grouping data members into their own class seems OK but I wonder what is the reasoning for heap allocating.